libraries/Microsoft.Bot.Builder/ActivityHandler.cs

```diff
@@ -434,13 +435,19 @@ protected virtual async Task<InvokeResponse> OnInvokeActivityAsync(ITurnContext<
         {
             try
             {
-                if (turnContext.Activity.Name == SignInConstants.VerifyStateOperationName || turnContext.Activity.Name == SignInConstants.TokenExchangeOperationName)
+                switch (turnContext.Activity.Name)
                 {
-                    await OnSignInInvokeAsync(turnContext, cancellationToken).ConfigureAwait(false);
-                    return CreateInvokeResponse();
-                }
+                    case SignInConstants.VerifyStateOperationName:
+                    case SignInConstants.TokenExchangeOperationName:
+                        await OnSignInInvokeAsync(turnContext, cancellationToken).ConfigureAwait(false);
+                        return CreateInvokeResponse();
+
+                    case "healthCheck":
```

Comment:
constant?

Reply:
we don't have constants for the other protocol words (for example TeamsActivityHandler) because this is the only place it is used (in our code base)

Reply:
The auth ones happen to have constants because we do use them in other places.

It wouldn't hurt to add constants for all of them, but I don't mind if we do that separately.

Reply:
This isn't the auth - it's the dispatching.

We don't have constants in the dispatching because its not application programming model, if it makes us feel better to have constants we can have internal constants - I don't know why we would do that but it's harmless. I wouldn't have thought if we want to do this then we should do it consistently. i.e. not part of this check-in. (perhaps a P3 issue for R10)